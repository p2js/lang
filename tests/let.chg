extern "intrinsic" fn bitcast<T, U>(v: T): U;

fn main(): i32 {
    let x: {x: i32} = bitcast(1);
    // ^ the initializer of this local is bitcast::<i32, ?0t> during type checking,
    // so we would fail in the property access below, however we can use the provided type annotation
    // to make it work.
    x.x;

    // ...but we still must prove that the initializer is a subtype. this should error:
    let y: i32 = 1i64;
    return y;
}
