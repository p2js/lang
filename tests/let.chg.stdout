fallback @ 4:31 4:32
expr @ 4:23 4:30 => extern "intrinsic" fn bitcast(...)
expr @ 4:31 4:32 => ?2t
expr @ 4:23 4:33 => ?1t
expr @ 8:5 8:6 => {x: i32}
expr @ 8:5 8:8 => i32
expr @ 11:18 11:19 => i64
expr @ 12:12 12:13 => i32
expr @ 12:5 12:13 => never
expr @ 3:16 14:1 => ()

